libr/cons/dietline.c

```diff
@@ -1196,12 +1255,394 @@ R_API const char *r_line_readline_cb_win(RLineReadCallback cb, void *user) {
 }
 #endif
 
+static inline void delete_next_char () {
+	if (I.buffer.length == I.buffer.index) {
+		I.buffer.data[I.buffer.index] = 0;
+		I.buffer.index = I.buffer.index > 0 ? I.buffer.index - 1 : 0;
+		I.buffer.length = I.buffer.length > 0 ? I.buffer.length - 1 : 0;
+		return;
+	}
+	memmove (I.buffer.data + I.buffer.index,
+		I.buffer.data + I.buffer.index + 1,
+		strlen (I.buffer.data + I.buffer.index + 1) + 1);
+}
+
+static inline void delete_prev_char () {
+	if (I.buffer.length == I.buffer.index) {
+		I.buffer.data[I.buffer.index] = 0;
+		I.buffer.index = I.buffer.index > 0 ? I.buffer.index - 1 : 0;
+		I.buffer.length = I.buffer.length > 0 ? I.buffer.length - 1 : 0;
+		return;
+	}
+	memmove (I.buffer.data + I.buffer.index - 1,
+		I.buffer.data + I.buffer.index,
+		strlen (I.buffer.data + I.buffer.index - 1) + 1);
+}
+
+static inline void delete_till_end () {
+	I.buffer.data[I.buffer.index] = '\0';
+	I.buffer.length = I.buffer.index;
+	I.buffer.index = I.buffer.index > 0 ? I.buffer.index - 1 : 0;
+}
+
+static void __print_prompt () {
+	int columns = r_cons_get_size (NULL) - 2;
+	int chars = R_MAX (1, strlen (I.buffer.data));	
+	int len, i, cols = R_MAX (1, columns - r_str_ansi_len (I.prompt) - 2);
+	r_cons_clear_line (0);
+	printf ("\r%s%s", Color_RESET, I.prompt);
+	fwrite (I.buffer.data, 1, R_MIN (cols, chars), stdout);
+	printf ("\r%s", I.prompt);
+	if (I.buffer.index > cols) {
+		printf ("< ");
+		i = I.buffer.index - cols;
+		if (i > sizeof (I.buffer.data)) {
+			i = sizeof (I.buffer.data) - 1;
+		}
+	} else {
+		i = 0;
+	}
+	len = I.buffer.index - i;
+	if (len > 0 && (i + len) <= I.buffer.length) {
+		fwrite (I.buffer.data + i, 1, len, stdout);
+	}
+	fflush (stdout);
+}
+
+static void __move_cursor_right () {
+	int utflen;
+#if USE_UTF8
+	{
+		char *s = I.buffer.data + I.buffer.index + 1;
+		utflen = 1;
+			while ((*s & 0xc0) == 0x80) {
```

Comment:
Indentation still broken here.

Reply:
I apologize for that, it seemed fine in vscode :/